### PR TITLE
[release 4.6] Bug 1891544: Adds ContainerRuntimeConfig gatherer

### PR DESCRIPTION
**Pull request by 0sewa0**
A backport (4.6) of a new data enhancement for gathering of ContainerRuntimeConfigs.
My cluster didn't have one by default so here is an example you can use to verify:
```yaml
apiVersion: machineconfiguration.openshift.io/v1
kind: ContainerRuntimeConfig
metadata:
 name: set-log-and-pid
spec:
 machineConfigPoolSelector:
   matchLabels:
     debug-crio: config-log-and-pid
 containerRuntimeConfig:
   pidsLimit: 2048
   logLevel: debug
```

## Categories

- [ ] Bugfix
- [X] Enhancement
- [X] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample archive

- `docs/insights-archive-sample/config/containerruntimeconfigs/set-log-and-pid.json`

## Documentation

- `docs/gathered-data.md`

## Unit Tests

- `pkg/gather/clusterconfig/clusterconfig_test.go`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## References

https://issues.redhat.com/browse/CCXDEV-3318
https://issues.redhat.com/browse/INSIGHTOCP-84
https://bugzilla.redhat.com/show_bug.cgi?id=1891544

### CODE REVIEW

**Comment by tremes** [comment]
Verified.

**Comment by sdodson** [comment]
(Patch Manager) This PR was reviewed for inclusion last week, however this adds feature work during the weeks immediately after release where we're focused on stability, upgrade success, and anything which blocks backports of critical customer impacting bugs. Since this doesn't meet those criteria it was not accepted this week and likely will not until we've been able to promote the 4.5 to 4.6 upgrade to the stable channel.

**Comment by sferich888** [comment]
@sdodson if PX approves backports like, this would that help unstick this PR's current state?

**Comment by tremes** [comment]
I've verified this again since it was some time ago. I created the `ContainerRuntimeConfig` above and it was gathered correctly in my cluster.
/lgtm

**Comment by sdodson** [comment]
Since this PR changed after `cherry-pick-approved` was granted it requires re-evaluation, what was the nature of the subsequent updates?

**Comment by 0sewa0** [comment]
@sdodson
The change was just a rebase.
The unfortunate thing that every time a data enhancement backport gets merged it will very likely cause a conflict with every other data enhancement backport(on the same version), this is obviously because that every thing is concentrated in ~2-3 files.
This issue was resolved in 4.7 due to a major refactor.

**Review by natiiix — APPROVED** [review]
/lgtm
(this has already been tested before, but needs a new lgtm label due to rebase)

**Pull request merged**
